src/multirust/command.rs

```diff
@@ -0,0 +1,109 @@
+use std;
+use std::env;
+use std::ffi::OsStr;
+use std::io::{self, Write};
+use std::path::PathBuf;
+use std::process::{self, Command, Output};
+use std::time::Instant;
+
+use Cfg;
+use errors::*;
+use multirust_utils;
+use telemetry;
+use telemetry::TelemetryEvent;
+
+
+pub fn run_command_for_dir<S: AsRef<OsStr>>(cmd: Command,
+                                            args: &[S],
+                                            cfg: &Cfg) -> Result<()> {
+    let arg0 = env::args().next().map(|a| PathBuf::from(a));
+    let arg0 = arg0.as_ref()
+        .and_then(|a| a.file_name())
+        .and_then(|a| a.to_str());
+    let arg0 = try!(arg0.ok_or(Error::NoExeName));
+    if (arg0 == "rustc" || arg0 == "rustc.exe") && cfg.telemetry_enabled() {
+        return telemetry_rustc(cmd, &args, &cfg);
+    }
+    
+    run_command_for_dir_without_telemetry(cmd, &args)
+}
+
+fn telemetry_rustc<S: AsRef<OsStr>>(cmd: Command, args: &[S], cfg: &Cfg) -> Result<()> {
+    let now = Instant::now();
+
+    let output = bare_run_command_for_dir(cmd, &args);
+
+    let duration = now.elapsed();
+
+    let ms = (duration.as_secs() as u64 * 1000) + (duration.subsec_nanos() as u64 / 1000 / 1000);
+
+    match output {
+        Ok(out) => {
+            let exit_code = out.status.code().unwrap_or(1);
+
+            let errors = match out.status.success() {
+                true => None,
+                _ => Some(String::from_utf8_lossy(&out.stderr).to_string()),
```

Comment:
Instead of capturing all of stderr, this needs to parse out strings that match `[Ennnn]` exactly and return a sorted and uniqified list of "Ennnn" strings (not integers - it's possible the "E" is significant). We can't return all of stderr because it contains identifiying info.